tests/snappi_tests/multidut/pfc/files/m2o_fluctuating_lossless_helper.py

```diff
@@ -0,0 +1,342 @@
+        prio_dscp_map (dict): Priority vs. DSCP map (key = priority)
+        snappi_extra_params (SnappiTestParams obj): additional parameters for Snappi traffic
+
+    Returns:
+        N/A
+    """
+    if snappi_extra_params is None:
+        snappi_extra_params = SnappiTestParams()
+
+    duthost1 = snappi_extra_params.multi_dut_params.duthost1
+    rx_port = snappi_extra_params.multi_dut_params.multi_dut_ports[0]
+    rx_port_id_list = [rx_port["port_id"]]
+    duthost2 = snappi_extra_params.multi_dut_params.duthost2
+    tx_port = [snappi_extra_params.multi_dut_params.multi_dut_ports[1],
+               snappi_extra_params.multi_dut_params.multi_dut_ports[2]]
+    tx_port_id_list = [tx_port[0]["port_id"], tx_port[1]["port_id"]]
+
+    pytest_assert(testbed_config is not None, 'Fail to get L2/3 testbed config')
+    stop_pfcwd(duthost1, rx_port['asic_value'])
+    disable_packet_aging(duthost1)
+    stop_pfcwd(duthost2, tx_port[0]['asic_value'])
+    disable_packet_aging(duthost2)
+
+    port_id = 0
+    # Generate base traffic config
+    snappi_extra_params.base_flow_config = setup_base_traffic_config(testbed_config=testbed_config,
+                                                                     port_config_list=port_config_list,
+                                                                     port_id=port_id)
+    __gen_traffic(testbed_config=testbed_config,
+                  port_config_list=port_config_list,
+                  rx_port_id_list=rx_port_id_list,
+                  tx_port_id_list=tx_port_id_list,
+                  pause_flow_name=PAUSE_FLOW_NAME,
+                  pause_prio_list=pause_prio_list,
+                  test_flow_name=TEST_FLOW_NAME,
+                  test_flow_prio_list=test_prio_list,
+                  test_flow_rate_percent=TEST_FLOW_AGGR_RATE_PERCENT,
+                  bg_flow_name=BG_FLOW_NAME,
+                  bg_flow_prio_list=bg_prio_list,
+                  bg_flow_rate_percent=BG_FLOW_AGGR_RATE_PERCENT,
+                  data_flow_dur_sec=DATA_FLOW_DURATION_SEC,
+                  data_pkt_size=DATA_PKT_SIZE,
+                  prio_dscp_map=prio_dscp_map)
+
+    flows = testbed_config.flows
+    all_flow_names = [flow.name for flow in flows]
+    data_flow_names = [flow.name for flow in flows if PAUSE_FLOW_NAME not in flow.name]
+
+    """ Run traffic """
+    flow_stats, switch_flow_stats = run_traffic(duthost=duthost1,
+                                                api=api,
+                                                config=testbed_config,
+                                                data_flow_names=data_flow_names,
+                                                all_flow_names=all_flow_names,
+                                                exp_dur_sec=DATA_FLOW_DURATION_SEC + DATA_FLOW_DELAY_SEC,
+                                                snappi_extra_params=snappi_extra_params)
+    flag = {'Test Flow': 'no_loss', 'Background Flow': {'loss': '10'}}
```

Comment:
can you provide some documentation on the structure of this flag dict

Reply:
description and an example has been given in  the function definition in traffic_generation.py file